### ISSUE TITLE
Running Qutip on Mac OS with Enthought Python 

### ISSUE THREAD

**Issue by rjmagyar** [qutip/qutip#382]
I just built and installed Qutip on the command line using Enthought python.  No errors...

Then when I run I get:

Enthought Canopy Python 2.7.9 | 64-bit | (default, May 20 2015, 20:01:56) 
[GCC 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2335.6)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from qutip import *
Compiling with an SDK that doesn't seem to exist: /Developer/SDKs/MacOSX10.6.sdk
Please check your Xcode installation
In file included from /usr/local/lib/gcc/x86_64-apple-darwin14.4.0/5.1.0/include-fixed/syslimits.h:7:0,
                 from /usr/local/lib/gcc/x86_64-apple-darwin14.4.0/5.1.0/include-fixed/limits.h:34,
                 from /Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/include/python2.7/Python.h:19,
                 from /Users/rjmagya/.pyxbld/temp.macosx-10.6-x86_64-2.7/pyrex/qutip/cy/spmatfuncs.c:16:
/usr/local/lib/gcc/x86_64-apple-darwin14.4.0/5.1.0/include-fixed/limits.h:168:61: error: no include path in which to search for limits.h
In file included from /Users/rjmagya/.pyxbld/temp.macosx-10.6-x86_64-2.7/pyrex/qutip/cy/spmatfuncs.c:16:0:
/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/include/python2.7/Python.h:33:19: fatal error: stdio.h: No such file or directory
compilation terminated.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/qutip/__init__.py", line 213, in <module>
    from qutip.qobj import *
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/qutip/qobj.py", line 60, in <module>
    from qutip.ptrace import _ptrace
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/qutip/ptrace.py", line 38, in <module>
    from qutip.sparse import sp_reshape
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/qutip/sparse.py", line 49, in <module>
    from qutip.cy.sparse_utils import (_sparse_profile, _sparse_permute,
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/qutip/cy/__init__.py", line 1, in <module>
    from qutip.cy.spmatfuncs import *
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pyximport/pyximport.py", line 431, in load_module
    language_level=self.language_level)
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pyximport/pyximport.py", line 209, in load_module
    inplace=build_inplace, language_level=language_level)
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pyximport/pyximport.py", line 186, in build_module
    reload_support=pyxargs.reload_support)
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/pyximport/pyxbuild.py", line 100, in pyx_to_dll
    dist.run_commands()
  File "/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 163, in run
    _build_ext.build_ext.run(self)
  File "/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/lib/python2.7/distutils/command/build_ext.py", line 337, in run
    self.build_extensions()
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/Cython/Distutils/build_ext.py", line 171, in build_extensions
    self.build_extension(ext)
  File "/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/lib/python2.7/distutils/command/build_ext.py", line 496, in build_extension
    depends=ext.depends)
  File "/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/lib/python2.7/distutils/ccompiler.py", line 574, in compile
    self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
  File "/Applications/Canopy.app/appdata/canopy-1.5.4.3105.macosx-x86_64/Canopy.app/Contents/lib/python2.7/distutils/unixccompiler.py", line 122, in _compile
    raise CompileError, msg
ImportError: Building module qutip.cy.spmatfuncs failed: ["CompileError: command 'gcc' failed with exit status 1\n"]
>>> import numpy as np
>>> 
>>> 
>>> from qutip import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/rjmagya/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/qutip/__init__.py", line 62, in <module>
    qutip.settings.ipython = False
AttributeError: 'module' object has no attribute 'settings'

######################

This seems similar to the build error in Installation on OS X with Homebrew script error #348.

Is this a bug in qutip ?  Is there a way to fix this?

**Comment by nonhermitian** [comment]
The canopy needs to be updated as it points toward an old os 10.6 whose SDK is not available on your newer osx. 

Paul

**Comment by ajgpitch** [comment]
@nonhermitian I am not a Mac man, so I am not sure how this would be done. @cgranade are you a Mac user?

**Comment by nonhermitian** [comment]
I wouldn't mess with it at all. Instead I would just get the anaconda package as it is the most up to date complete package and is guaranteed to work.

**Comment by cgranade** [comment]
I'm not a Mac user, no. I run a Linux desktop and a Windows 10 laptop. That said, helping colleagues deal with OS X / Enthought Canopy compatibility issues is a big part of the reason that I started using Anaconda instead. They seem to do a much better job at dealing with Xcode updates erasing old SDKs.

**Comment by arnelg** [comment]
+1 for anaconda over enthought on mac (mac user here).

**Comment by ajgpitch** [comment]
Looks like Anaconda is a winner most typically

**Comment by nonhermitian** [comment]
Closing this as well.

**Comment by sahmed95** [comment]
An alternative solution if one does not wish to use Anaconda can be to change the Python version in use to a user installed one instead of the default version supplied by macOS. I have discussed it in #348 

It is perhaps because the default Python version does not detect the local python packages (numpy/scipy). After this fix, I was able to do a `sudo -H python setup.py install` and everything worked fine for me.
